An 170 coordinate tx wallet support

sucildossj opened the pull request:
Included following changes

scantxoutset rpc changes for asset
listunspent rpc changes for asset
create premium transaction rpc
send asset transaction rpc
send pegout transaction rpc

sucildossj committed:
fix: include changes for include asset diff in scantxoutset

sucildossj committed:
change wallet listunspent api for asset and preconf support

sucildossj committed:
fix: include reserve support on listunspent

sucildossj committed:
feat: include send pegout implementation

sucildossj committed:
updating changes related to wallet rpc

jasonribble reviewed (approved):
Looks good. Feel free to keep formatting from upstream to reduce merge conflict in the future

sucildossj committed:
include asset id in coutput point

sucildossj committed:
change asset id as class instead for int64_t

sucildossj committed:
remove blocktype for casset class since asset creation only belong to raw block

sucildossj committed:
include CAsset in coutpoint and updated testcase

jasonribble reviewed (approved):
The `CAsset` in the outpoint looks looks good

sucildossj commented:
> The `CAsset` in the outpoint looks looks good
Currently, I’m changing the data type to a buffer that holds 8 bytes (int64_t) for the block number and 2 bytes (int16_t) for the asset number within the block. Each block can have only 256 new assets, with asset IDs ranging from 001 to 256. We need to filter the block template to allow only 256 assets per block and ensure this rule is validated in the consensus layer as well. @jasonribble

sucildossj commented:
Additionally, if an asset creation transaction is still in the mempool, we need to wait until it is confirmed before allowing any transfers.

sucildossj committed:
including asset id implementation

sucildossj committed:
included changes releated to asset outpoint changes

-- validated asset id from blockchain
-- getblocktemplate included only 256 asset creation
-- consensus validation for 256 asset creation

sucildossj committed:
refactor: disable asset creationg and transfer from mempool

sucildossj committed:
remove asset support on preconf transaction

sucildossj committed:
feat: include utxo fee logic for asset type transaction only

sucildossj committed:
refactor: fixed unit testing changes for block filter and gensis block hash changes

sucildossj committed:
include unit test changes after transaction template change

sucildossj committed:
included changes on getrawtransaction for asset id

sucildossj committed:
fix: include diffcult nan fixes on getblockheader

sucildossj committed:
including functional testcase hrp changes

sucildossj committed:
include asset doc for scantxoutset

sucildossj committed:
changed primitive class structure based on node

sucildossj committed:
updating changes for merkle root changes

sucildossj committed:
include asset transaction test file in repo

sucildossj committed:
update value on supply and controller output

sucildossj committed:
enable preconf mempool for testcase and include testcase for preconf
